### PR TITLE
crypto, keystore: encrypt/decrypt private keys, read/writes to file

### LINKED ISSUE #366: Keystore: ability to create, export, import keys
We need a keystore to store node keys. The keystore should be able to generate new keys, export keys to an encrypted file, and import keys (raw or from encrypted file) 

Need to determine method of encryption.

### PR DESCRIPTION
**Pull request by noot**
## Changes
- add interfaces for keys to crypto package, will be needed for using both ed25519/sr25519 keys.
- add keystore package which can encrypt/decrypt data w/ a password using AES
- implemented functions to encrypt/decrypt keys and read/write the data to a file

## Tests:
```
go test ./crypto/... ./keystore/...
```

### Issues:
closes #366

### CODE REVIEW

File: crypto/ed25519_test.go [hunk 1]
```diff
@@ -3,6 +3,8 @@ package crypto
 import (
 	"reflect"
 	"testing"
+
+	ed25519 "golang.org/x/crypto/ed25519"
```
Comment (ansermino) [inline crypto/ed25519_test.go]: This is now `crypto/ed25519`. See second paragraph: https://godoc.org/golang.org/x/crypto/ed25519

https://godoc.org/crypto/ed25519

File: crypto/ed25519.go [hunk 2]
```diff
@@ -36,40 +41,90 @@ func GenerateEd25519Keypair() (*Ed25519Keypair, error) {
...
+	return kp.Public()
+}
+
+// Encode returns the bytes underlying the Ed25519PrivateKey
+func (k *Ed25519PrivateKey) Encode() []byte {
+	return []byte(ed25519.PrivateKey(*k))
+}
+
+// Decode turns the input bytes into a Ed25519PrivateKey
+// the input must be 32 bytes, or the function will return an error
```
Comment (dutterbutter) [inline crypto/ed25519.go]: 32 bytes or 64 bytes? 

`NewEd25519PrivateKey` checks for 64 bytes...

File: crypto/ed25519_test.go (hunk 1 above)
Comment (noot) [reply crypto/ed25519_test.go]: true, I will change it

File: crypto/ed25519.go (hunk 2 above)
Comment (noot) [reply crypto/ed25519.go]: you are right, I've updated the comment

**Pull request merged**